episodes/anonymization.md

```diff
@@ -32,19 +33,352 @@ Metadata elements in imaging, such as patient names and addresses, are often cle
+    image2_z = (0,largest_component_mask.GetSize()[2]-1),
+    image1_npa = fixed(sitk.GetArrayViewFromImage(brain_only)),
+    image2_npa = fixed(sitk.GetArrayViewFromImage(largest_component_mask)),
+    title1=fixed("new image"),
+    title2=fixed("mask")
+    )
+
+
+```
+```output
+```
+
+![Our partial soft tissue stripping.](fig/our_cca_stripped.png){alt='Home-made deface'}
+
+
+ Now you may notice that we put together and erosion and dilation, and we could have used instead a closure. There are many ways to solve this problem. An entirely different approach would be to use grow-from seed as below (not this may take a while to execute):
+
+ ```python
+ def guess_seed_point(img):
+    """
+    This function guesses a seed point for the brain in the middle of the image, and returns some seed points.
+    """
+    possible_point = round(img.GetSize()[0]/2), round(img.GetSize()[1]/2), round(img.GetSize()[2]/2)
+    # Get the pixel value at the potential location
+    pixel_value = img.GetPixel(*possible_point)
+    if pixel_value > 0:
+        picked_point = possible_point
+    else:
+        # just move over a bit and hope for better
+        new_possible_point = round(img.GetSize()[0]/2) + round(img.GetSize()[0]/10) , round(img.GetSize()[1]/2), round(img.GetSize()[2]/2)
+        picked_point = new_possible_point
+    return picked_point
+# do some reality check of a look at the value in your seed point  
+seed_point = guess_seed_point(sag_image) 
+pixel_value = sag_image.GetPixel(seed_point)
+print(pixel_value)
+ ```
+
+ ```output
+ 394.37042236328125
+ ```
+
+Grow from seed, then display:
+
+
+ ```python
+seed_point = guess_seed_point(sag_image)  
+lower_threshold = 370   # lower threshold
+upper_threshold = 480   # upper threshold
+
+
+seed_mask = sitk.ConnectedThreshold(
+    sag_image, seedList=[seed_point],
+    lower=lower_threshold,
+    upper=upper_threshold)
+# let's dilate up a bit
+seed_mask= sitk.BinaryDilate(seed_mask, [5, 5, 5])
+
+# apply the mask to the image
+brain_only = sitk.Mask(sag_image, seed_mask)
+# display to see what happened
+interact(
+    display_images,
+    image1_z=(0, sag_image.GetSize()[2] - 1),
+    image2_z=(0, brain_only.GetSize()[2] - 1),
+    image1_npa=fixed(sitk.GetArrayViewFromImage(sag_image)),
+    image2_npa=fixed(sitk.GetArrayViewFromImage(brain_only)),
+    title1=fixed("Original"),
+    title2=fixed("Seeded and Masked")
+)
+ ```
+
+
+ ```output
+```
+
+![Our grown from seed soft tissue stripping.](fig/grown_from_seed.png){alt='Home-made deface by grow from seed'}
+
+ In the two solutions above we made masks. They each have different problems we can see. For example in the grown from seed images we stripped out part of the genu and splenium of the corpus callosum not to mention the entire pons. An alternative is that could also have used a registered atlas as a mask. Regardless of the base algorithms we begin with, if we want a better image, we can keep tinkering with our algorithms.
```

Comment:
I really like how you're doing a home-made one here first before showing the ready-made ones, because the latter will also be imperfect and it's important that people realise this. Making them play with an algorithm of their own is the perfect way of giving them a feel for the limitations.